test/AbstractTransitionController.spec.ts

```diff
@@ -0,0 +1,209 @@
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import 'chai/register-should';
+import sinonChai from 'sinon-chai';
+import { Animation, TimelineMax } from 'gsap';
+import { render, getComponent, getApp } from './util/index.spec';
+import { TransitionId } from '../test/util/component/ChildComponentA/ChildComponentATransitionController';
+import TimelineType from '../src/lib/enum/TimelineType';
+import TransitionDirection from '../src/lib/enum/TransitionDirection';
+chai.use(sinonChai);
+
+describe('#AbstractTransitionController.spec', () => {
+  let component;
+  let app;
+
+  before(() => {
+    render();
+    app = getApp();
+    component = getComponent('child-component-a');
+  });
+
+  describe('transitionIn', () => {
+    it('should transition in the component', () => {
+      return component.transitionController.transitionIn()
+        .then(() => expect(component.transitionController.isHidden).to.be.false);
+    });
+  });
+
+  describe('transitionOut', () => {
+    it('should transition out the component', () => {
+      return component.transitionController.transitionIn(true)
+        .then(() => component.transitionController.transitionOut(true))
+        .then(() => expect(component.transitionController.isHidden).to.be.true);
+    });
+  });
+
+  describe('startLoopingAnimation', () => {
+    it('should start the looping animation with exact parameters', () =>{
+      const reset = false;
+      const label = TransitionId.LOOP_1;
+
+      const spy = sinon.spy(component.transitionController, 'setupTimeline');
+      expect(component.transitionController.startLoopingAnimation(label, reset)).to.be.undefined;
+      expect(spy).to.be.calledWithExactly(TimelineType.LOOPING, reset, label);
+      spy.restore();
+    });
+  });
+
+  describe('setupTimeline', () => {
+    it('should setup looping timeline with label', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.LOOPING, false, TransitionId.LOOP_1)).to.be.undefined;
+    });
+
+    it('should setup looping timeline', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.LOOPING, false)).to.be.undefined;
```

Comment:
What are we testing here in these tests? Does checking undefined really check to correct workingof the function?